pkg/koordlet/metricsadvisor/collectors/performance/performance_collector_linux.go

```diff
@@ -198,25 +194,43 @@ func (p *performanceCollector) collectContainerPSI() {
 			containerStatusesMap[containerStat] = meta
 		}
 	}
+
+	metricsChan := make(chan metriccache.MetricSample, len(containerStatusesMap)*2)
+	psiMetrics := make([]metriccache.MetricSample, 0)
+
+	go p.getContainerPSIMetrics(containerStatusesMap, metricsChan)
+
+	for item := range metricsChan {
+		psiMetrics = append(psiMetrics, item)
+	}
+
+	// save container's psi metrics to tsdb
+	p.saveMetric(psiMetrics)
+
+	p.started.Store(true)
+	klog.V(5).Infof("collectContainerPSI for time window %s finished at %s, container num %d",
+		timeWindow, time.Now(), len(containerStatusesMap))
+}
+
+func (p *performanceCollector) getContainerPSIMetrics(containerStatusesMap map[*corev1.ContainerStatus]*statesinformer.PodMeta, metricChan chan<- metriccache.MetricSample) {
+	defer close(metricChan)
+
 	var wg sync.WaitGroup
 	wg.Add(len(containerStatusesMap))
 	metrics.ResetContainerPSI()
+	collectTime := time.Now()
 	for containerStatus, podMeta := range containerStatusesMap {
 		pod := podMeta.Pod
 		cgroupDir := podMeta.CgroupDir
 		go func(parentDir string, status *corev1.ContainerStatus, pod *corev1.Pod) {
 			defer wg.Done()
-			p.collectSingleContainerPSI(parentDir, status, pod)
+			p.collectSingleContainerPSI(parentDir, status, pod, collectTime, metricChan)
```

Comment:
Similarly, we can let getContainerPSIMetrics return a slice and append here, so dose the pod PSI